components/sync/driver/brave_sync_service_impl.cc

```diff
@@ -44,6 +45,11 @@ void BraveSyncServiceImpl::Initialize() {
     StopAndClear();
     brave_sync_prefs_.SetSyncV1Migrated(true);
   }
+
+  // P3A ping for those who have sync disabled
+  if (!user_settings_->IsFirstSetupComplete()) {
+    base::UmaHistogramExactLinear("Brave.Sync.Status.2", 0, 3);
```

Comment:
will this overwrite a value recorded in `OnDeviceInfoChange()`?
1. Enable sync, 1 or 2 is recorded in OnDeviceInfoChange
2. Restart the browser: recorded 0 and that's it?

Reply:
if sync is still enabled, then after restart of browser `user_settings_->IsFirstSetupComplete()` will return true, and the statement `base::UmaHistogramExactLinear("Brave.Sync.Status.2", 0, 3);` will not be executed

Reply:
the short answer is - no, it will not overwrite the value with the steps from above

Reply:
gotcha. Do we have any sync-related tests that could be expanded with checking this metric? You might use`base::HistogramTester` to quickly check if something was recorded.

Reply:
Maybe `BraveSyncServiceImplTest` if it uses non-mocked `BraveSyncServiceImplDelegate` . I will look more.